recipes/boost/all/conanfile.py

```diff
@@ -256,6 +262,9 @@ def configure(self):
         if self.settings.compiler == "Visual Studio" and "MT" in str(self.settings.compiler.runtime) and self.options.shared:
             raise ConanInvalidConfiguration("Boost can not be built as shared library with MT runtime.")
 
+        if self.settings.compiler == "Visual Studio" and self.options.shared and self.options.numa:
+            raise ConanInvalidConfiguration("Cannot build a shared boost with numa support on Visual Studio")
```

Comment:
@prince-chrismc 
The limitation to shared is intentional.
dll's don't allow missing symbols, static libraries allow it.
This means a static numa library can be built, but not a shared library.

I said: it can be built. I didn't say that it will work :smile: